### ISSUE TITLE
dotnet publish does not package like DeployOnBuild=true does for SDK-style project

### ISSUE THREAD

**Issue by jviau** [Azure/azure-webjobs-sdk#2319]
Please provide a succinct description of the issue.

#### Repro steps

1. Create a new SDK-style webjobs project with webjob properties set. i.e.:
``` xml
    <IsWebJobProject>true</IsWebJobProject>
    <WebJobName>MyWebJob</WebJobName>
    <WebJobType>Continuous</WebJobType>
```

2. Run `dotnet publish`

#### Expected behavior

The web job is packaged into the correct structure: `app_data\Jobs\Continuous\MyWebJob\...`

#### Actual behavior

The web job is not packaged into the correct structure. It is only possible to get the correct structure via `dotnet build -p:DeployOnBuild=true`. This means you do not have support for things like `dotnet publish --no-build`. In fact I do not know a way to package a webjob _without_ triggering the build target. This makes it very difficult to split the build & package steps into two distinct msbuild calls (to differentiate them on CI machines, and to allow for non-msbuild operations to happen between build and publish - such as signing).

#### Known workarounds

It is not even possible to chain `dotnet publish` into the web job package flow as target `_CopyWebJobFilesToIntermediateOutputPath` depends on `Publish`, so it results in circular references (as `Publish` target will be your entry). At least I could not find a way to make this work.
The only alternative is to define another Target as such:

``` XML
  <Target Name="Package" DependsOnTargets="$(DotNetPublishDependsOn)" />
```

Then you can run `dotnet msbuild /t:Package` and adding `/p:NoBuild=true` to skip the build. But this does not allow for the use of the `dotnet publish` commands.

#### Related information 

Provide any related information 

* This is _not_ using `Microsoft.Web.WebJobs.Publish`, but `Microsoft.NET.Sdk.Publish`
* Using .netcore sdk 2.2.106

**Comment by brettsam** [comment]
@fabiocav / @mathewc -- is this where we track issues with WebJobs builds? I'm not sure where the tooling for this lives...

**Comment by fabiocav** [comment]
The tooling is not maintained on GitHub, but @vijayrkn might be able to comment on the above (sorry about the tagging you on such an old issue).

**Comment by fabiocav** [comment]
@jviau can you please confirm you're still experiencing this issue?

**Comment by vijayrkn** [comment]
This should be fixed in functions sdk 3.0.9
